### ISSUE TITLE
[mlir] mlir-opt asserts when dumping the pass pipeline

### ISSUE THREAD

**Issue by ggeorgakoudis** [llvm/llvm-project#90344]
Running `mlir-opt` with these options:
```shell
mlir-opt --sparsifier="enable-runtime-library=false parallelization-strategy=dense-outer-loop gpu-triple=nvptx64-nvidia-cuda gpu-chip=sm_70" --dump-pass-pipeline
```
asserts with:
```console
builtin.module(func.func(linalg-generalize-named-ops),sparsification-and-bufferization,sparse-storage-specifier-to-llvm,func.func(canonicalize{  max-iterations=10 max-num-rewrites=-1 region-simplify=true test-convergence=false top-down=true}),func.func(finalizing-bufferize),sparse-gpu-codegen{enable-runtime-library=true num-threads=1024},gpu.module(strip-debuginfo),gpu.module(convert-scf-to-cf),gpu.module(convert-gpu-to-nvvm{has-redux=false index-bitwidth=0 use-bare-ptr-memre
f-call-conv=false}),func.func(convert-linalg-to-loops),func.func(convert-vector-to-scf{full-unroll=false lower-tensors=false target-rank=1}),func.func(expand-realloc{emit-deallocs=true}),func.func(convert-scf-to-cf),expand-strided-metadata,lower-affine,convert-vector-to-llvm{enable-amx=false enable-arm-neon=false enable-arm-sve=false enable-x86vector=false force-32bit-vector-indices=true reassociate-fp-reductions=false},finalize-memref-to-llvm{index-bitwidth=0 use-aligned-all
oc=false use-generic-functions=false},func.func(convert-complex-to-standard),func.func(arith-expand{include-bf16=false}),func.func(convert-math-to-llvm{approximate-log1p=true}),convert-math-to-libm,convert-complex-to-libm,convert-vector-to-llvm{enable-amx=false enable-arm-neon=false enable-arm-sve=false enable-x86vecto
r=false force-32bit-vector-indices=true reassociate-fp-reductions=false},convert-complex-to-llvm,convert-vector-to-llvm{enable-amx=false enable-arm-neon=false enable-arm-sve=false enable-x86vector=false force-32bit-vector-indices=true reassociate-fp-reductions=false},convert-func-to-llvm{index-bitwidth=0 use-bare-ptr-m
emref-call-conv=false},nvvm-attach-target{O=2 chip=sm_70 fast=false features= ftz=false  module= triple=nvptx64-nvidia-cuda},gpu-to-llvm{gpu-binary-annotation=gpu.binary use-bare-pointers-for-host=false use-bare-pointers-for-kernels=false},gpu-module-to-binary{format=fatbin handler=unknown data value for option
UNREACHABLE executed at /usr/workspace/ggeorgak/projects/mfem/llvm-project/mlir/include/mlir/Pass/PassOptions.h:136!
```

Some option of `gpu-module-to-binary` does not parse.

git describe: `llvmorg-19-init-8111-ga27ab3f54377`

Reproduction: https://godbolt.org/z/zaajfonEv

**Comment by joker-eph** [comment]
@fabianmcg : seems like an option you added?

**Comment by fabianmcg** [comment]
Yes, the issue here is that `handler` doesn't have a textual representation (it's an attribute), hence it crashes when dumping the options. Here's a minimal reproducer:
https://godbolt.org/z/rfdY6hMv3
 I'll fix it.

**Comment by joker-eph** [comment]
All pass options must round-trip really!

**Comment by fabianmcg** [comment]
Yeah I missed it. My plan for fixing it is adding option parsers for `Attribute` and `Type`, @joker-eph  thoughts?

**Comment by joker-eph** [comment]
It's possible, we'll have to escape the string if you use the string serialization, but that should work I think?

**Comment by fabianmcg** [comment]
Yeah, I think it should work.  I'll create the PR.

**Comment by fabianmcg** [comment]
Fixed in #90368
